server/matchmaker/algorithm.py

```diff
@@ -118,4 +150,11 @@ def _rank_all(searches: List[Search]) -> Dict[Search, List[Search]]:
 
 
 def _rank_partners(search: Search, others: Iterable[Search]) -> List[Search]:
-    return heapq.nlargest(SM_NUM_TO_RANK, others, key=lambda other: search.quality_with(other))
+    return heapq.nlargest(
+        SM_NUM_TO_RANK, 
+        filter(
```

Comment:
Ok, so the only problem with this is it makes the debugging output kindof pointless. Because before it would print out all of the qualities and thresholds between people and you could see why they weren't matching. Now people don't match and there's no debugging output to show for it. If they are going to match anyways, then it prints out their qualities.